docs/recipes/UsingImmutableJS.md

```diff
@@ -0,0 +1,218 @@
+# Using Immutable.js with Redux
+## Table of Contents
+- [Why should I use an immutable-focused library such as Immutable.JS?](#why-use-immutable-library)
+- [Why should I choose Immutable.JS as an immutable library?](#why-choose-immutable-js)
+- [What are the issues with using Immutable.JS?](#issues-with-immutable-js)
+- [Is Immutable.JS worth the effort?](#is-immutable-js-worth-effort)
+- [What are the Recommended Best Practices for using Immutable.JS with Redux?](#immutable-js-best-practices)
+
```

Comment:
Can we throw in a caveat paragraph somewhere here towards the front that says something along the lines of: "_If_ you choose to use Immutable.js, here are some (opinionated) best practices that you may find useful"?  I always get nervous about people taking stuff in docs as a "must" or "absolute requirement", and just want to keep it clear that people aren't _required_ to do exactly what this page says.